guides/source/active_record_validations.md

```diff
@@ -777,28 +908,56 @@ class Person < ApplicationRecord
 end
 ```
 
-WARNING. Note that some databases are configured to perform case-insensitive
-searches anyway.
+WARNING: Some databases are configured to perform case-insensitive searches
+anyway.
+
+A `:conditions` option can be used to specify additional conditions as a `WHERE`
+SQL fragment to limit the uniqueness constraint lookup:
 
-There is a `:conditions` option that you can specify additional conditions as a
-`WHERE` SQL fragment to limit the uniqueness constraint lookup (e.g.
-`conditions: -> { where(status: 'active') }`).
+```ruby
+validates :name, uniqueness: { conditions: -> { where(status: 'active') } }
+```
 
 The default error message is _"has already been taken"_.
 
 See [`validates_uniqueness_of`][] for more information.
 
-[`validates_uniqueness_of`]: https://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_uniqueness_of
-[`add_index`]: https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_index
-[the MySQL manual]: https://dev.mysql.com/doc/refman/en/multiple-column-indexes.html
+[`validates_uniqueness_of`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_uniqueness_of
+[`add_index`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_index
+[the MySQL manual]:
+    https://dev.mysql.com/doc/refman/en/multiple-column-indexes.html
 [the MariaDB manual]: https://mariadb.com/kb/en/compound-composite-indexes/
-[the PostgreSQL manual]: https://www.postgresql.org/docs/current/static/ddl-constraints.html
+[the PostgreSQL manual]:
+    https://www.postgresql.org/docs/current/static/ddl-constraints.html
+
+### `validate`
+
+The `validate` validator is used with `belongs_to` and `has_one` associations to
+automatically validate the associated object whenever the parent object is
+saved. This option ensures that any new or updated associated objects are
+validated before the parent object can be successfully saved. By default, this
+option is set to `false`, meaning associated objects will not be automatically
+validated.
+
+```ruby
+class Post < ApplicationRecord
+  belongs_to :author, validate: true
+end
+```
+
+In this example, every time a `Post` is saved, the associated `Author` will also
+be validated. If the `Author` is invalid, the `Post` will not be saved.
+
+If you need to ensure that associated objects are validated every time the
+parent object is updated, consider using `validates_associated`.
```

Comment:
Argh, this option opens another can of worms 😬 

The `validate` options isn't really a validator, so I don't think it belongs here.
Also [the logic ](https://github.com/rails/rails/blob/ac98964910369dd9821316a68f5477bd0d8e4ced/activerecord/lib/active_record/reflection.rb#L700-L711)of when a record is/isn't validated is pretty complex 😞 
And autosave has [more corner cases](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/autosave_association.rb)

I don't think this is properly documented anywhere (both in the API docs and the guides), so I think it's good to have it documented somewhere. Maybe we can add another section to this guide? Or maybe it belongs in the associations guide?
I've added a rough outline.
````markdown

Automatic Association Validations
---------------------------------

Active Record enables validations on some associations.

### `belongs_to` Validations

By default `belongs_to` associations validate the presence of the associated
record to guarantee reference consistency every time the record is saved.

```ruby
class Post < ApplicationRecord
  belongs_to :author
end
```

```irb
irb> post = Post.new
irb> post.valid?
=> false # Author is missing
```

Set the `:optional` option to true if this is not what you want. This means
that the foreign key in one table might not reliably point to a valid primary
key in the referenced table.

```ruby
class Post < ApplicationRecord
  belongs_to :author, optional: true
end
```

### `has_many` Validations

For `has_many` and `has_many :through` associations the associated records will
be validated every time the parent is saved.

```
class Post < ActiveRecord::Base
  belongs_to :author
  validates :title, presence: true
end

class Author < ActiveRecord::Base
  has_many :posts
end
```

```irb
irb> author = Author.new
irb> post = author.posts.new
irb> author.save
=> false
irb> post.title = "Some title"
irb> author.save
=> true

### `validate` Option

The `validate` option is used with `belongs_to` and `has_one` associations to
automatically validate the associated object whenever the parent object is
saved. This option ensures that any new associated objects are
validated before the parent object can be successfully saved. By default, this
option is not set, meaning associated objects will not be automatically
validated.

```ruby
class Post < ApplicationRecord
  belongs_to :author, validate: true
end
```

In this example, every time a `Post` is saved, the associated `Author` will also
be validated. If the `Author` is invalid, the `Post` will not be saved.

If you need to ensure that associated objects are validated every time the
parent object is updated, consider using `validates_associated`.

### `autosave` Option

When the `autosave` option is set to true, the associated records will be
validated everytime the parent record is saved, instead of just when creating.

```ruby
class Supplier < ApplicationRecord
  has_one :account, autosave: true
end

class Account < ApplicationRecord
  validates :terms, presence: true
end
```

```irb
irb> supplier = Supplier.first
irb> supplier.account.terms = nil
irb> supplier.save
=> false # Account is invalid
```
````

Reply:
Association guide makes sense

Reply:
@p8 as we discussed offline - looking at the section I want to try a table approach in the association guides which I think would be easier to digest but I suspect that it will need a few iterations so maybe a separate PR is better.  I've removed the original section here and created a task for it on our board.